### ISSUE TITLE
Wordlist: consecutive dupe suppression without rules is gone

### ISSUE THREAD

**Issue by magnumripper** [magnumripper/JohnTheRipper#1012]
We've had it working, but now that code is just gone. Simplest test is just `./john -stdin -stdout` so we don't involve any memory buffering or such things.

**Comment by magnumripper** [comment]
It was dropped in 15949eb (July 2014) because it caused nodes to get out of sync when truncation caused a dupe. That code was introduced in 11e5191 (Feb 2012) before which we also lacked dupe suppression.

Core has it in dummy_rules_apply():
```
static char *dummy_rules_apply(char *word, char *rule, int split, char *last)
{
        word[length] = 0;
        if (strcmp(word, last))
                return strcpy(last, word);
        return NULL;
}
```
Jumbo's version just returns word. I wonder why, I'm trying to track changes.

**Comment by magnumripper** [comment]
OK, we do not want that strcmp and strcpy in there because they are not needed when running buffered. So it was moved to a non-buffered and non-rule specific code path but there was this bug with nodes getting out of sync and this was fixed the wrong way in 15949eb.

**Comment by magnumripper** [comment]
I reverted 15949eb and hopefully fixed it correctly (like in b575eec5). I did check that this did not reopen #699 but we should test it more.
